packages/xdl/src/createWebpackCompiler.js

```diff
@@ -0,0 +1,166 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import chalk from 'chalk';
+import formatWebpackMessages from 'react-dev-utils/formatWebpackMessages';
+import clearConsole from 'react-dev-utils/clearConsole';
+import * as ProjectUtils from './project/ProjectUtils';
+const CONSOLE_TAG = 'webpack';
+
+function log(projectRoot, message, showInDevtools = true) {
+  if (showInDevtools) {
+    ProjectUtils.logInfo(projectRoot, CONSOLE_TAG, message);
+  } else {
+    console.log(message);
+  }
+}
+
+function logWarning(projectRoot, message) {
+  ProjectUtils.logWarning(projectRoot, CONSOLE_TAG, message);
+}
+
+function logError(projectRoot, message) {
+  ProjectUtils.logError(projectRoot, CONSOLE_TAG, message);
+}
+
+function printInstructions(projectRoot, appName, urls, showInDevtools) {
+  const _log = message => log(projectRoot, message, showInDevtools);
+
+  console.log('');
+  _log(`You can now view ${chalk.bold(appName)} in the browser.`);
+  console.log('');
```

Comment:
I think these console.logs should be removed. Add newlines inside the `_log` message as necessary.

Reply:
but this will add new lines to the devtools. Doing it this way only adds newlines to the terminal

Reply:
My point is that everything should go in a single log message with some line breaks in it. Only then it will look the same in Dev Tools and the terminal.

You can do it like this:
```js
const message = `You can now view ${chalk.bold(appName)} in the browser.\n\n`;
if (urls.lanUrlForTerminal) {
  message += `  ${chalk.bold('Local:')}            ${urls.localUrlForTerminal}\n`;
  message += `  ${chalk.bold('On Your Network:')}  ${urls.lanUrlForTerminal}\n`
} else {
  message += `  ${urls.localUrlForTerminal}\n`;
}
```
Then you also won't need to log those empty strings to the console.